src/components/opening-hours/opening-hours.vue

```diff
@@ -68,6 +76,49 @@ export default {
   },
   computed: {
     ...mapState(['isMobile']),
+    timeSlots() {
+      const { openingHoursSpace, openingHoursBuilding } = this
+      const newOpeningHours = openingHoursSpace.map((openingHour, index) => {
+        const buildingStart = openingHoursBuilding[index].time.length ? openingHoursBuilding[index].time[0][0] : []
+        const buildingEnd = openingHoursBuilding[index].time.length ? openingHoursBuilding[index].time[0][1] : []
+        const newTimeslots = []
+
+        newTimeslots.day = openingHour.day
+
+        newTimeslots.times = openingHour.time.reduce((timeslots, timeslot, index) => {
+          // Check if there is a busy slot before the first open time slot
+          if (index === 0 && new Date(timeslot[0]) > new Date(buildingStart)) {
+            const start = buildingStart
+            const end = timeslot[0]
+            timeslots.push({ time: [start, end], busy: false })
+          }
+
+          // Always add open time slot to array
+          timeslots.push({ time: timeslot, busy: false })
+
+          // Check if there is a busy slot between to open time slots
+          const nextTimeslot = openingHour.time[index + 1]
+          if (nextTimeslot && new Date(nextTimeslot[0]) > new Date(timeslot[1])) {
+            const start = timeslot[1]
+            const end = nextTimeslot[0]
+            timeslots.push({ time: [start, end], busy: true })
+          }
+
+          // Check if there is a busy slot after the last open time slot
+          if (index === 6 && new Date(buildingEnd) > new Date(timeslot[1])) {
```

Comment:
I think, right now we're only checking if the 7th slot of a space has later closing hours than the last available slot ending. I guess `index` should be `openingHour.time.length - 1`